scripts/limits.js

```diff
@@ -12,142 +12,65 @@ const {
-const network = process.env.NETWORK || 'localnet';
-let nonce = 0;
-let packageId;
-let gateway;
-let discovery;
-let gatewayInfo = {};
-const discoveryInfo = {};
-
-function calculateNextSigners(n = 3, threshold = 2) {
-    signerKeys = Array.from({length: n}, getRandomBytes32);
-    const pubKeys = signerKeys.map((key) => Secp256k1Keypair.fromSecretKey(arrayify(key)).getPublicKey().toRawBytes());
-    const keys = signerKeys.map((key, index) => {
-        return { privKey: key, pubKey: pubKeys[index] };
-    });
-    keys.sort((key1, key2) => {
-        for (let i = 0; i < 33; i++) {
-            if (key1.pubKey[i] < key2.pubKey[i]) return -1;
-            if (key1.pubKey[i] > key2.pubKey[i]) return 1;
-        }
-
-        return 0;
-    });
-    return {
-        signerKeys: keys.map((key) => key.privKey),
-        signers: {
-            signers: keys.map((key) => {
-                return { pub_key: key.pubKey, weight: 1 };
-            }),
-            threshold,
-            nonce: hexlify([++nonce]),
-        },
-    }
-}
-
-async function rotateSigners(n = 3, threshold = 2) {console.log(n, threshold);
-    let proofSigners = gatewayInfo.signers;
-    let proofKeys = gatewayInfo.signerKeys;
-    let nextSigners = calculateNextSigners(n, threshold);
-
-    const encodedSigners = WeightedSigners.serialize(nextSigners.signers).toBytes();
-
-    const hashed = hashMessage(encodedSigners, COMMAND_TYPE_ROTATE_SIGNERS);
-
-    const message = MessageToSign.serialize({
-        domain_separator: domainSeparator,
-        signers_hash: keccak256(WeightedSigners.serialize(proofSigners).toBytes()),
-        data_hash: hashed,
-    }).toBytes();
-
-    const signatures = signMessage(proofKeys.slice(0, proofSigners.threshold), message);
-    const encodedProof = Proof.serialize({
-        signers: proofSigners,
-        signatures,
-    }).toBytes();
-
-    const builder = new TxBuilder(client);
-
-    await builder.moveCall({
-        target: `${packageId}::gateway::rotate_signers`,
-        arguments: [gateway, CLOCK_PACKAGE_ID, encodedSigners, encodedProof],
-    });
-
-    await builder.signAndExecute(keypair);
-    gatewayInfo = {
-        ...gatewayInfo,
-        ...nextSigners,
-    };
-    console.log(gatewayInfo, nextSigners);
-}
+async function main() {
+    const COMMAND_TYPE_APPROVE_MESSAGES = 0;
+    const COMMAND_TYPE_ROTATE_SIGNERS = 1;
+
+    let client;
+    const operator = Ed25519Keypair.fromSecretKey(arrayify(getRandomBytes32()));
+    const deployer = Ed25519Keypair.fromSecretKey(arrayify(getRandomBytes32()));
+    const keypair = Ed25519Keypair.fromSecretKey(arrayify(getRandomBytes32()));
+    const domainSeparator = getRandomBytes32();
+    const network = process.env.NETWORK || 'localnet';
+    let nonce = 0;
+    let packageId;
+    let gateway;
+    let discovery;
+    let gatewayInfo = {};
+    const discoveryInfo = {};
+
+    function calculateNextSigners(n = 3, threshold = 2) {
```

Comment:
Helpers should still be defined outside and relevant args should be passed in, so it's clear what this function depends on, rather than implicitly capturing variables

Reply:
changed everything again